src/config.py

```diff
@@ -0,0 +1,115 @@
+        utils.validate_key(data["Source"]["Share"], "TargetKmsKey", "TargetKmsKey")
+        if data["Source"]["Share"].get("AssumeRole") is None:
+            data["AssumeSourceRole"] = None
+        else:
+            LOGGER.info("Assuming source AWS role")
+            data["AssumeSourceRole"] = utils.assume_aws_role(
+                data["Source"]["Share"]["AssumeRole"]
+            )
+    return data
+
+
+def load_target(data):
+    utils.validate_key(data, "Target", "Target")
+    utils.validate_key(data["Target"], "DBIdentifier", "DBIdentifier")
+
+    # if data["Target"].get("MasterUserPassword") is None:
+    #     data["Target"]["MasterUserPassword"] = generate_password()
+    if data["Target"].get("AssumeRole") is None:
+        data["AssumeTargetRole"] = None
+    else:
+        LOGGER.info("Assuming target AWS role")
+        data["AssumeTargetRole"] = utils.assume_aws_role(data["Target"]["AssumeRole"])
+
+    utils.validate_key(data["Target"], "DBSubnetGroupName", "DBSubnetGroupName")
+    utils.validate_key(data["Target"], "VpcSecurityGroupIds", "VpcSecurityGroupIds")
+    utils.validate_key(data["Target"], "Tags", "Tags")
+    utils.validate_key(data["Target"], "DBInstanceClass", "DBInstanceClass")
+
+    if data["Target"].get("CopyTagsToSnapshot") is None:
+        data["Target"]["CopyTagsToSnapshot"] = True
+    if data["Target"].get("PubliclyAccessible") is None:
+        data["Target"]["PubliclyAccessible"] = False
+    if data["Target"].get("DeleteExistingTarget") is None:
+        data["Target"]["DeleteExistingTarget"] = False
+
+    return data
+
+
+def load_config(filename):
+    if not is_config_exists(filename):
+        LOGGER.error("Provided configuration file is not available")
+        exit(1)
+
+    with open(filename) as file:
+        data = json.load(file)
+
+    if data.get("ClusterMode") is None:
+        data["ClusterMode"] = True
+
+    data = load_source(data)
+    data = load_target(data)
+    return data
+
+
+def is_config_exists(filename):
+    return os.path.isfile(filename)
+
+
+def is_sharing_enabled(data):
+    if data["Source"].get("Share") is None:
+        return False
+    return data["Source"]["Share"]
+
+
+def parse_config(value, tf_outputs, assume_role):
+    if type(value) == str:
+        pattern = re.compile(r"\${(.+):(.+)}")
+        result = pattern.search(value)
+        if result is None:
+            return value
+        if result.group(1) == "tf":
+            if tf_outputs.get(result.group(2)) is None:
+                LOGGER.error("%s does not exists in TF state" % result.group(2))
+            value = tf_outputs[result.group(2)]
+        elif result.group(1) == "ssm":
+            value = ssm.get_parameter(assume_role, result.group(2))
+        elif result.group(1) == "env":
+            if os.environ.get(result.group(2)) is None:
+                LOGGER.error(
+                    "%s environment variable does not exists" % result.group(2)
+                )
```

Comment:
I would avoid nested if blocks of combination of for/if being nested.  Makes code hard to read, troubleshoot and maintain